consensus/state.go

```diff
@@ -1117,18 +1117,23 @@ func (cs *State) defaultDecideProposal(height int64, round int32) {
 		cs.Logger.Error("enterPropose: Error signing proposal", "height", height, "round", round, "err", err)
 	}
 
+	// TODO(ismail): capture this in the Consensus ADR
 	// post data to ipfs
 	// TODO(evan): don't hard code context and timeout
-	if cs.ipfs != nil {
-		// longer timeouts result in block proposers failing to propose blocks in time.
-		ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond*1500)
-		defer cancel()
-		// TODO: post data to IPFS in a goroutine
-		err := ipld.PutBlock(ctx, cs.ipfs.Dag(), block)
-		if err != nil {
-			cs.Logger.Error(fmt.Sprintf("failure to post block data to IPFS: %s", err.Error()))
-		}
+	//
+	// longer timeouts result in block proposers failing to propose blocks in time.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond*1500)
+	defer cancel()
+	cs.Logger.Info("Putting Block to ipfs", "height", block.Height)
+	// TODO: post data to IPFS in a goroutine
+	err = ipld.PutBlock(ctx, cs.ipfs.Dag(), block)
+	if err != nil {
+		// If PutBlock fails we will be the only node that has the data
+		// this means something is seriously wrong and we can not recover
+		// from that automatically.
+		panic(fmt.Sprintf("failure to post block data to IPFS: %s", err.Error()))
```

Comment:
Maybe just skip our leadership turn? Panicking is too brutal for consensus. Although, we can consider this code as not final, but keep the panic here for us to catch such issue earlier and remove it in long term

Reply:
I actually agree with the comment in lines 1131-1133, and think we should end up keeping this panic.

Reply:
Yeah, I think we should shut down instead of keeping a node around that can't propose properly when it is their turn. Otherwise, it will cause the network to start a new round each time its this node's turn.